flutter_web_auth_2/android/src/main/kotlin/com/linusu/flutter_web_auth_2/CallbackActivity.kt

```diff
@@ -1,20 +1,54 @@
 package com.linusu.flutter_web_auth_2
 
+import android.annotation.SuppressLint
 import android.app.Activity
+import android.content.Intent
 import android.net.Uri
+import android.os.Build
 import android.os.Bundle
 
-class CallbackActivity: Activity() {
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
+class CallbackActivity : Activity() {
 
-    val url = intent?.data
-    val scheme = url?.scheme
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
 
-    if (scheme != null) {
-      FlutterWebAuth2Plugin.callbacks.remove(scheme)?.success(url.toString())
+        val url = intent?.data ?: fixAutoVerifyNotWorks(intent)
+        val scheme = url?.scheme
+
+        if (scheme != null) {
+            FlutterWebAuth2Plugin.callbacks.remove(scheme)?.success(url.toString())
+        }
+        @SuppressLint("ObsoleteSdkInt")
+        if (Build.VERSION.SDK_INT >= 21) {
+            finishAndRemoveTask()
+        } else {
+            finish()
+        }
+    }
+
+
+    /** Fix sometimes android:autoVerify="true" cannot works when it can't access Google after installation.
+     * See https://stackoverflow.com/questions/76383106/auto-verify-not-always-working-in-app-links-using-android
+     *
+     * must register in AndroidManifest.xml :
+     * <intent-filter>
+     *     <action android:name="android.intent.action.SEND" />
+     *     <category android:name="android.intent.category.DEFAULT" />
+     *     <data android:mimeType="text/plain" />
+     *</intent-filter>
```

Comment:
Could you add proper documentation in the README maybe? For example, in the Setup/Android section?
If it is not mandatory to do this, please properly point that out

Reply:
Sometimes it may open a browser that doesn't support customtab, and need to add this to receive sharing from the browser.